### PR TITLE
Refactored component settings and torch device

### LINKED ISSUE #303: Refactor SimulationModule
Currently the `SimulationModule` class is defined in the `__init__.py` file under `simpa/core`. However, in a structural perspective it would make sense to separate core methods like `run` from simulation module specific methods such as `get_additional_flags`. 

Suggested solution:
Create a `CoreModule` class in the `core/__init__.py` and move the `SimulationModule` to `core/simulation_modules/__init__.py` and inherit from `CoreModule`.

### PR DESCRIPTION
**Pull request by TomTomRixRix**
**Please check the following before creating the pull request (PR):**
- [x] Did you run automatic tests?
As described in #307 this automatic test fails locally, but should work on github.

- [x] Did you run manual tests?
~~We noticed that the acoustic_forward_model test yield wrong results. The directions of the reconstructed images are flipped. This is most likely not related to our code changes, but rather to #270. This should be addressed together with #163.~~
**Edit**: This was only a local configuration error with a wrong virtual environment still having the old SIMPA version
 
- [ ] Is the code provided in the PR still backwards compatible to previous SIMPA versions?
It's complicated. It's backwards compatible if you only simulate a pipeline. But once you import `SimulationModules` you have to adjust the path from `.core` to `.core.simulation_modules`. So this mostly affects developers of new adapters. 

 **List any specific code review questions**
Are there more things to refactor into PipelineModules or SimulationModules ? 

 **Provide issue / feature request fixed by this PR**
 Fixes #303

### CODE REVIEW

File: simpa/core/simulation_modules/__init__.py [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
 # SPDX-License-Identifier: MIT
+
+from abc import abstractmethod
+
+from simpa.core import PipelineModule
+from simpa.utils import Settings
+
+class SimulationModule(PipelineModule):
+    """
+    Defines a simulation module that is a step in the simulation pipeline.
```
Comment (kdreher) [inline simpa/core/simulation_modules/__init__.py:12]: Please add that a `SimulationModule` is one of these 4: Volume Creation, Light Propagation Modeling, Acoustic Wave Propagation Modeling, Image Reconstruction

File: simpa/core/simulation_modules/__init__.py [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+        """
+        Called before running the run method from PipelineModule
+        """
+        self.logger.info(f"Starts running {self.__class__.__name__}")
+
+    def after_running(self):
+        """
+        Called after running the run method from PipelineModule
+        """
+        self.logger.info(f"Ended running {self.__class__.__name__}")
```
Comment (kdreher) [inline simpa/core/simulation_modules/__init__.py:42]: What are these functions used for?
You didn't change any other adapter that uses them?

**Review by kdreher — CHANGES_REQUESTED** [review]

File: simpa/core/simulation_modules/__init__.py (hunk 2 above)
Comment (kdreher) [reply simpa/core/simulation_modules/__init__.py:42]: Also, please make sure that the base classes for all the devices are also defined in the `__init__.py`

Comment (TomTomRixRix) [reply simpa/core/simulation_modules/__init__.py:42]: Agree, currently these methods are not used. Might be interesting for the future but for now we removed them

**Comment by TomTomRixRix** [comment]
We've now updated the PR @kdreher

File: simpa/utils/utils.py
Comment (kdreher) [inline simpa/utils/utils.py:1]: Please move this functionality to the [calculate.py](https://github.com/IMSY-DKFZ/simpa/blob/main/simpa/utils/calculate.py) file. It is a collection of helper functions that do some kind of mathematical calculation.

But if this is not used in the simpa package and only for testing, there are already some [test_utils](https://github.com/IMSY-DKFZ/simpa/blob/main/simpa_tests/test_utils/__init__.py)

**Review by kdreher — CHANGES_REQUESTED** [review]
Looks good :) Just move the `are_equal` function and then we'll merge

**Review by kdreher — APPROVED** [review]
Looks good

**Pull request merged**